cmd/cmd.go

```diff
@@ -919,27 +932,40 @@ func NewCLI() *cobra.Command {
 		CompletionOptions: cobra.CompletionOptions{
 			DisableDefaultCmd: true,
 		},
-		Version: version.Version,
+		PersistentPreRunE: func(cmd *cobra.Command, _ []string) error {
+			if !slices.Contains([]string{"serve", "version"}, cmd.Name()) {
+				return checkServerHeartbeat(cmd, nil)
```

Comment:
Abstracting this into the root command makes it more difficult for someone to reason about what's happening in each individual command, and it really only saves a few lines of code.

Reply:
On top of reducing duplication, this is less error prone than requiring each subcommand to register their own PreRun. Only a small subset of commands should _not_ check server heartbeat. This changes the check to an opt-out operation rather than an opt-in operation which is less error prone.

Reply:
I'd rather be explicit here and than implicitly calling it. It's a single line per command.